Source/PhysicalParticleContainer.cpp

```diff
@@ -1426,6 +1433,164 @@ PhysicalParticleContainer::Evolve (int lev,
+                        psplit_uy.push_back( uyp[i] );
+                        psplit_uz.push_back( uzp[i] );
+                        psplit_w.push_back( wp[i]/np_split );
+                    }
+                }
+#elif (AMREX_SPACEDIM==3)
+		if (split_type==0){
+		    // Split particle in two along each axis
+		    // 6 particles in 2d
+		    for (int ishift = -1; ishift < 2; ishift +=2 ){
+			for (int jshift = -1; jshift < 2; jshift +=2 ){
+			    for (int kshift = -1; kshift < 2; kshift +=2 ){
+				// Add one particle with offset in x, y and z
+				psplit_x.push_back( xp[i] + ishift*dx[0]/2 );
+				psplit_y.push_back( yp[i] + jshift*dx[1]/2 );
+				psplit_z.push_back( zp[i] + jshift*dx[2]/2 );
+				psplit_ux.push_back( uxp[i] );
+				psplit_uy.push_back( uyp[i] );
+				psplit_uz.push_back( uzp[i] );
+				psplit_w.push_back( wp[i]/np_split );
+			    }
+			}
+		    }
+		} else {
+		    // Split particle in two along each diagonal
+		    // 8 particles in 3d
+                    for (int ishift = -1; ishift < 2; ishift +=2 ){
+                        // Add one particle with offset in x
+                        psplit_x.push_back( xp[i] + ishift*dx[0]/2 );
+                        psplit_y.push_back( yp[i] );
+                        psplit_z.push_back( zp[i] );
+                        psplit_ux.push_back( uxp[i] );
+                        psplit_uy.push_back( uyp[i] );
+                        psplit_uz.push_back( uzp[i] );
+                        psplit_w.push_back( wp[i]/np_split );
+                        // Add one particle with offset in y
+                        psplit_x.push_back( xp[i] );
+                        psplit_y.push_back( yp[i] + ishift*dx[1]/2 );
+                        psplit_z.push_back( zp[i] );
+                        psplit_ux.push_back( uxp[i] );
+                        psplit_uy.push_back( uyp[i] );
+                        psplit_uz.push_back( uzp[i] );
+                        psplit_w.push_back( wp[i]/np_split );
+                        // Add one particle with offset in z
+                        psplit_x.push_back( xp[i] );
+                        psplit_y.push_back( yp[i] );
+                        psplit_z.push_back( zp[i] + ishift*dx[2]/2 );
+                        psplit_ux.push_back( uxp[i] );
+                        psplit_uy.push_back( uyp[i] );
+                        psplit_uz.push_back( uzp[i] );
+                        psplit_w.push_back( wp[i]/np_split );
+                    }
+		}
+#endif
+		// invalidate the particle
+                p.m_idata.id = -p.m_idata.id;
+            }
+        }
+    }
+    if (np_split_to_add>0){
+	// Add local arrays psplit_x etc. to the temporary
+	// particle container pctmp_split. Split particles
+	// are tagged with p.id()=NoSplitParticleID so that 
+	// they are not re-split when entering a higher level
+	// AddNParticles calls Redistribute, so that particles
+	// in pctmp_split are in the proper grids and tiles
+	pctmp_split.AddNParticles(lev, 
```

Comment:
This needs to be called on all processes. Maybe some of the don't have any to add? But if some procs call Redistribute and some don't, that could cause a hang.